simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp

```diff
@@ -205,183 +140,142 @@ VehicleKinematicState comToBaselink(const VehicleKinematicState & com)
+  const auto pub_sub_node = std::make_shared<PubSubNode>();
 
-    // go backward
-    _resetInitialpose();
-    _sendBwdGear();
-    _sendCommand(cmdGen(sim_node->now(), 0.0f, -target_vel, -target_acc));
-    isOnBackward(*(pub_sub_node->current_state_), init_state);
+  const float32_t target_vel = 5.0f;
+  const float32_t target_acc = 5.0f;
+  const float32_t target_steer = 0.2f;
 
-    // go forward left
-    _resetInitialpose();
-    _sendFwdGear();
-    _sendCommand(cmdGen(sim_node->now(), target_steer, target_vel, target_acc));
-    isOnForwardLeft(*(pub_sub_node->current_state_), init_state);
+  auto _resetInitialpose = [&]() { resetInitialpose(sim_node, pub_sub_node); };
+  auto _sendFwdGear = [&]() { sendGear(GearCommand::DRIVE, sim_node, pub_sub_node); };
+  auto _sendBwdGear = [&]() { sendGear(GearCommand::REVERSE, sim_node, pub_sub_node); };
+  auto _sendCommand = [&](const auto & _cmd) { sendCommand(_cmd, sim_node, pub_sub_node); };
 
-    // go backward right
-    _resetInitialpose();
-    _sendBwdGear();
-    _sendCommand(cmdGen(sim_node->now(), -target_steer, -target_vel, -target_acc));
-    isOnBackwardRight(*(pub_sub_node->current_state_), init_state);
+  // check pub-sub connections
+  {
+    size_t expected = 1;
+    EXPECT_EQ(pub_sub_node->pub_ackermann_command_->get_subscription_count(), expected);
+    EXPECT_EQ(pub_sub_node->pub_gear_cmd_->get_subscription_count(), expected);
+    EXPECT_EQ(pub_sub_node->pub_initialpose_->get_subscription_count(), expected);
+    EXPECT_EQ(pub_sub_node->current_odom_sub_->get_publisher_count(), expected);
   }
 
+  // check initial pose
+  _resetInitialpose();
+  const auto init_state = *(pub_sub_node->current_odom_);
+
+  // go forward
+  _resetInitialpose();
+  _sendFwdGear();
+  _sendCommand(cmdGen(sim_node->now(), 0.0f, 0.0f, target_vel, target_acc, 0.0f));
+  isOnForward(*(pub_sub_node->current_odom_), init_state);
+
+  // go backward
+  // NOTE: positive acceleration with reverse gear drives the vehicle backward.
+  _resetInitialpose();
+  _sendBwdGear();
+  _sendCommand(cmdGen(sim_node->now(), 0.0f, 0.0f, -target_vel, target_acc, 0.0f));
+  isOnBackward(*(pub_sub_node->current_odom_), init_state);
+
+  // go forward left
+  _resetInitialpose();
+  _sendFwdGear();
+  _sendCommand(cmdGen(sim_node->now(), target_steer, 0.0f, target_vel, target_acc, 0.0f));
+  isOnForwardLeft(*(pub_sub_node->current_odom_), init_state);
+
+  // go backward right
+  // NOTE: positive acceleration with reverse gear drives the vehicle backward.
+  _resetInitialpose();
+  _sendBwdGear();
+  _sendCommand(cmdGen(sim_node->now(), -target_steer, 0.0f, -target_vel, target_acc, 0.0f));
+  isOnBackwardRight(*(pub_sub_node->current_odom_), init_state);
+
   rclcpp::shutdown();
 }
 
 // Send a control command and run the simulation.
 // Then check if the vehicle is moving in the desired direction.
-TEST(test_simple_planning_simulator, test_moving_ackermann)
+TEST(TestSimplePlanningSimulator, TestIdealSteerVel)
```

Comment:
As an alternative, you can use the parameterized test (`TEST_P`).
Reference: https://zxc-programming.hatenablog.com/entry/2018/11/15/134540

Reply:
FYI: When using pytest, you can also use parameterized tests. https://docs.pytest.org/en/latest/how-to/parametrize.html

Reply:
This is very useful. I'll take it. Thank you!

Reply:
Addressed in 
https://github.com/autowarefoundation/autoware.universe/pull/422/commits/2321d7677ff6f98678648ac0020d7930b831ede1

Reply:
Great!!